FirebasePerformance/Sources/AppActivity/FPRAppActivityTracker.m

```diff
@@ -121,6 +150,55 @@ - (FIRTrace *)activeTrace {
   return self.backgroundSessionTrace;
 }
 
++ (BOOL)isPrewarmAvailable {
+  if (![[GULAppEnvironmentUtil applePlatform] isEqualToString:@"ios"]) {
+    return NO;
+  }
+  NSString *systemVersion = [GULAppEnvironmentUtil systemVersion];
+  if ([systemVersion length] == 0) {
+    return NO;
+  }
+  NSArray *versionComponents = [systemVersion componentsSeparatedByString:@"."];
+  NSInteger majorVersion = [versionComponents[0] integerValue];
+  return majorVersion >= 15;
+}
+
+- (BOOL)isActivePrewarmEnabled {
+  return ([self.configurations prewarmDetectionMode] == 1 ||
+          [self.configurations prewarmDetectionMode] == 3);
+}
+
+- (BOOL)isDoubleDispatchEnabled {
+  return ([self.configurations prewarmDetectionMode] == 2 ||
+          [self.configurations prewarmDetectionMode] == 3);
+}
+
+- (BOOL)isApplicationPreWarmed {
+  if (![FPRAppActivityTracker isPrewarmAvailable]) {
+    return NO;
+  }
+
+  // Force drop all events according to RC flag
+  if ([self.configurations prewarmDetectionMode] == 0) {
+    return YES;
+  }
+
+  NSDictionary<NSString *, NSString *> *environment = [NSProcessInfo processInfo].environment;
+  if ([self isActivePrewarmEnabled] && environment[@"ActivePrewarm"] != nil &&
```

Comment:
Checking the environment variable seem to be happening more than once. Is there a way to abstract this?